### PR TITLE
Update 283-Move-Zeroes.js

### PR DESCRIPTION
**Pull request by aadil42**
Added solution with linear space just for reference's sake.

File(s) Modified: 283-Move-Zeroes.js
Language(s) Used: JavaScript

### CODE REVIEW

File: javascript/283-Move-Zeroes.js [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+ */
+var moveZeroes = function(nums) {
+
+    const zeroAtTheEnd = Array(nums.length).fill(0);
+    let left = 0;
+    let right = zeroAtTheEnd.length - 1;
+
+    for (let i = 0; i < nums.length; i++) {
+        if (nums[i] === 0) {
+            right--;
```
Comment (aakhtar3) [inline javascript/283-Move-Zeroes.js:15]: Add continue an remove else branch

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/283-Move-Zeroes.js (hunk 1 above)
Comment (aadil42) [reply javascript/283-Move-Zeroes.js]: @aakhtar3, I updated the code, we don't need else or the continue part now.

File: javascript/283-Move-Zeroes.js [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+ * https://leetcode.com/problems/move-zeroes/
+ * @param {number[]} nums
+ * @return {void} Do not return anything, modify nums in-place instead.
+ */
+var moveZeroes = function(nums) {
+
+    const zeroAtTheEnd = Array(nums.length).fill(0);
+    let left = 0;
+    for (let i = 0; i < nums.length; i++) {
+        if (nums[i]) {
```
Comment (aakhtar3) [inline javascript/283-Move-Zeroes.js:13]: You should define your if conditions

File: javascript/283-Move-Zeroes.js [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+ * Time Complexity  O(N) | Space Complexity O(N);
+ * https://leetcode.com/problems/move-zeroes/
+ * @param {number[]} nums
+ * @return {void} Do not return anything, modify nums in-place instead.
+ */
+var moveZeroes = function(nums) {
+
+    const zeroAtTheEnd = Array(nums.length).fill(0);
+    let left = 0;
+    for (let i = 0; i < nums.length; i++) {
```
Comment (aakhtar3) [inline javascript/283-Move-Zeroes.js:12]: Most of the solutions for 2 pointer and cyclic sort are in while loop. Easier to remember that pattern over for loop

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/283-Move-Zeroes.js (hunk 2 above)
Comment (aadil42) [reply javascript/283-Move-Zeroes.js:13]: @aakhtar3, What do you mean by defining if conditions?

File: javascript/283-Move-Zeroes.js (hunk 3 above)
Comment (aakhtar3) [reply javascript/283-Move-Zeroes.js:12]: Look at the Space O(1) if condition

File: javascript/283-Move-Zeroes.js (hunk 2 above)
Comment (aakhtar3) [reply javascript/283-Move-Zeroes.js:13]: Look at the Space O(1) if condition

File: javascript/283-Move-Zeroes.js (hunk 3 above)
Comment (aadil42) [reply javascript/283-Move-Zeroes.js:12]: Hey @aakhtar3, check now. Is this ok?

Comment (aakhtar3) [reply javascript/283-Move-Zeroes.js:12]: Let's do this
```js
const arr = new Array(nums.length).fill(0);

let [ left, right ] = [ 0, 0 ];

while (right < nums.length) {
	const isZero = (nums[right] === 0);
	if (!isZero) {
		arr[left] = nums[right];
		left++;
	}

	right++;
}

return arr;
```

Comment (aadil42) [reply javascript/283-Move-Zeroes.js:12]: That could work, let me try to submit it on leetCode.

Comment (aadil42) [reply javascript/283-Move-Zeroes.js:12]: Oh, shoot! we won't be able to because we're creating a new array.

Comment (aadil42) [reply javascript/283-Move-Zeroes.js:12]: But I did try some test cases and it seems to run just fine. I'll update the code.

Comment (aadil42) [reply javascript/283-Move-Zeroes.js]: Please check now @aakhtar3

File: javascript/283-Move-Zeroes.js [hunk 4]
```diff
@@ -1,3 +1,29 @@
+/**
+ * Linear Time
```
Comment (aakhtar3) [inline javascript/283-Move-Zeroes.js:2]: Two Pointer

File: javascript/283-Move-Zeroes.js [hunk 5]
```diff
@@ -1,3 +1,29 @@
...
+ * Time Complexity  O(N) | Space Complexity O(N);
```
Comment (aakhtar3) [inline javascript/283-Move-Zeroes.js:3]: Time O(N) | Space O(N)

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/283-Move-Zeroes.js (hunk 4 above)
Comment (aadil42) [reply javascript/283-Move-Zeroes.js:2]: done

File: javascript/283-Move-Zeroes.js (hunk 5 above)
Comment (aadil42) [reply javascript/283-Move-Zeroes.js:3]: done

**Comment by aadil42** [comment]
@aakhtar3, I update the file with the suggested changes. Please have a look.

**Review by aakhtar3 — APPROVED** [review]

**Pull request merged**